mento-protocol/mento-core
# feat: proxy ChainlinkRelayerFactory

### Description

This PR sets up upgradeability for the ChainlinkRelayerFactory (see #476). 

TODO:

* [x] integration test of the ChainlinkRelayerFactory

### Tested

Integration tests that:

- Setup a proxied ChainlinkRelayerFactory
- Ensure the TransparentUpgradeableProxy/ProxyAdmin pattern works correctly
- Sets up a new rate feed with an enabled circuit breaker
- Tests that price reports coming from the relayer correctly trigger/reset the circuit breaker.

### Related issues

- Fixes #459
